Add Redlite tasks for safety benchmarking

inno-simon opened the pull request:
Hello,

I've added a red teaming oriented group of tasks that have been featured in [this paper](https://arxiv.org/abs/2404.09785)

I've omitted the tasks based on `realtoxicity` as the original dataset is already featured in `lm_eval`.  

The PR includes 7 tasks designed to evaluate various safety aspects of an LLM, as well as a custom metric, `best-of-pem-rouge` which takes computes both a rouge score, and a "prefix exact match" and picks the highest value.  Details and justifications can be found in the paper. 

To test the integration, i ran all the tasks on two models, one local (gemma 2B) and one online (gpt4).  I compared these results with the ones from the paper (which can also be found [here](https://innodata-llm-safety.innodata.com/#/)) and it produced the following :

![image](https://github.com/EleutherAI/lm-evaluation-harness/assets/104781596/57e653e0-e22c-4f34-84a9-e6cb555dcc50)

There are small differences, for which there are three likely explanations : 
1. `redlite` is leveraging the `system prompt` part of the datasets when the model supports it; `lm_eval` does not
2. the general non-deterministic nature of LLMs
3. possible discrepancy in model parameters used (temperature, etc.)

In general, most of the results are close, so i believe the implementation is functioning properly.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2020) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2020) before we can accept your contribution.<br/><hr/>**Simon** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2020) it.</sub>

haileyschoelkopf commented on lm_eval/tasks/redlite/utils.py:
```diff
@@ -0,0 +1,103 @@
+
+
+def calculate_rouge(gold: str, pred: str) -> float:
+    scorer = rouge_scorer.RougeScorer(["rouge2"], use_stemmer=True)
```
Could we carry over the change made in #2090 to prevent excessive logs from RougeScorer?

inno-simon replied:
done

haileyschoelkopf commented on lm_eval/tasks/redlite/README.md:
```diff
@@ -0,0 +1,54 @@
+# Task-name
```
Replace placeholder

inno-simon replied:
done

haileyschoelkopf commented on lm_eval/tasks/redlite/README.md:
```diff
@@ -0,0 +1,54 @@
+
+#### Groups
+
+* `group_name`: `redlite`
```
This group does not seem to be defined anywhere?

inno-simon replied:
removed

haileyschoelkopf commented on lm_eval/tasks/redlite/rt-cogensumm.yaml:
```diff
@@ -0,0 +1,12 @@
+task: rt-cogensumm
```
Regarding task naming: could we prefix `redlite_` to all these tasks' names for clarity?

(most tasks in the repo also tend to favor `_` instead of `-`, but that's less important)

inno-simon replied:
done

haileyschoelkopf reviewed (changes requested):
Thank you for the PR!

Assuming you and the other authors of this paper are good with the implementation here, then this looks good to me aside from a couple nits:

- could the small differences between this implementation and the one in the paper be documented in the task's README.md, perhaps?
- one small change to make ROUGE logs mostly go away, following #2090 
- Some minor changes to make tasks' naming more clear.

inno-simon commented:
Addressed comments and reran each task for sanity; obtained same numbers.
Added `Caveats` section to README
Other contributors to the paper have reviewed the implementation as well.
